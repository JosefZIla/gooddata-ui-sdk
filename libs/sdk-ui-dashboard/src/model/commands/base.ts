// (C) 2021 GoodData Corporation
/**
 * All available command types.
 *
 * @internal
 */
export type DashboardCommandType =
    | "GDC.DASHBOARD.CMD.LOAD"
    | "GDC.DASHBOARD.CMD.SAVE"
    | "GDC.DASHBOARD.CMD.SAVEAS"
    | "GDC.DASHBOARD.CMD.RESET"
    | "GDC.DASHBOARD.CMD.RENAME"
    | "GDC.DASHBOARD.CMD.AF.ADD"
    | "GDC.DASHBOARD.CMD.AF.REMOVE"
    | "GDC.DASHBOARD.CMD.AF.MOVE"
    | "GDC.DASHBOARD.CMD.AF.CHANGE_SELECTION"
    | "GDC.DASHBOARD.CMD.AF.SET_PARENT";
/**
 * Base type for all commands.
 *
 * Commands are dispatched using dispatcher obtained by useDashboardDispatcher(). All the dispatchers are 'contextual' -
 * they target backend, workspace and dashboard in depending on the Dashboard component tree from which the dispatch
 * is done.
 *
 * @internal
 */
export interface IDashboardCommand {
    /**
     * Command type. Always starts with "GDC.DASHBOARD.CMD"
     */
    readonly type: DashboardCommandType;

    /**
     * Correlation ID can be provided when creating a command. All events emitted during the command processing
     * will contain the same correlation ID.
     *
     * If the correlation ID is not specified, a random string will be assigned.
     */
    readonly correlationId?: string;
}
