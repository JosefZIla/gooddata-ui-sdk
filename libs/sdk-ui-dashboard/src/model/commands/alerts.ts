// (C) 2021-2022 GoodData Corporation

import { IWidgetAlert, IWidgetAlertDefinition } from "@gooddata/sdk-backend-spi";
import { ObjRef } from "@gooddata/sdk-model";
import { IDashboardCommand } from "./base";

/**
 * Payload of the {@link CreateAlert} command.
 * @alpha
 */
export interface CreateAlertPayload {
    /**
     * The alert to be created.
     */
    readonly alert: IWidgetAlertDefinition;
}

/**
 * Creates Kpi alert.
 *
 * @alpha
 */
export interface CreateAlert extends IDashboardCommand {
    readonly type: "GDC.DASH/CMD.ALERT.CREATE";
    readonly payload: CreateAlertPayload;
}

/**
 * Creates the SaveAlert command. Dispatching this command will result in the creating Kpi alert on the backend.
 *
 * @param alert - specify alert to create.
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 * @alpha
 */
export function createAlert(alert: IWidgetAlertDefinition, correlationId?: string): CreateAlert {
    return {
        type: "GDC.DASH/CMD.ALERT.CREATE",
        correlationId,
        payload: {
            alert,
        },
    };
}

//
//
//

/**
 * Payload of the {@link UpdateAlert} command.
 * @alpha
 */
export interface UpdateAlertPayload {
    /**
     * The alert to be updated.
     */
    readonly alert: IWidgetAlert;
}

/**
 * Updates Kpi alert.
 *
 * @alpha
 */
export interface UpdateAlert extends IDashboardCommand {
    readonly type: "GDC.DASH/CMD.ALERT.UPDATE";
    readonly payload: UpdateAlertPayload;
}

/**
 * Creates the UpdateAlert command. Dispatching this command will result in the updating Kpi alert on the backend.
 *
 * @param alert - specify alert to update.
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing

 * @alpha
 */
export function updateAlert(alert: IWidgetAlert, correlationId?: string): UpdateAlert {
    return {
        type: "GDC.DASH/CMD.ALERT.UPDATE",
        correlationId,
        payload: {
            alert,
        },
    };
}

//
//
//

/**
 * Payload of the {@link RemoveAlerts} command.
 * @alpha
 */
export interface RemoveAlertsPayload {
    /**
     * References to Kpi alerts that should be removed.
     */
    readonly refs: ObjRef[];
}

/**
 * Removes Kpi alerts.
 *
 * @alpha
 */
export interface RemoveAlerts extends IDashboardCommand {
    readonly type: "GDC.DASH/CMD.ALERTS.REMOVE";
    readonly payload: RemoveAlertsPayload;
}

/**
 * Creates the RemoveAlerts command. Dispatching this command will result in the removing Kpi alerts on the backend.
 *
 * @param refs - specify ObjRef of the alerts to remove
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing

 * @alpha
 */
export function removeAlerts(refs: ObjRef[], correlationId?: string): RemoveAlerts {
    return {
        type: "GDC.DASH/CMD.ALERTS.REMOVE",
        correlationId,
        payload: {
            refs,
        },
    };
}
