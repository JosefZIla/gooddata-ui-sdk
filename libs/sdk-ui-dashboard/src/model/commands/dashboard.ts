// (C) 2021 GoodData Corporation

import { DashboardConfig } from "../types/commonTypes";
import { IWorkspacePermissions } from "@gooddata/sdk-backend-spi";

/**
 * All available command types.
 *
 * @internal
 */
export type DashboardCommandType = "GDC.DASHBOARD.CMD.LOAD";

/**
 * Base type for all commands.
 *
 * Commands are dispatched using dispatcher obtained by useDashboardDispatcher(). All the dispatchers are 'contextual' -
 * they target backend, workspace and dashboard in depending on the Dashboard component tree from which the dispatch
 * is done.
 *
 * @internal
 */
export interface IDashboardCommand {
    /**
     * Command type. Always starts with "GDC.DASHBOARD.CMD"
     */
    type: DashboardCommandType;

    /**
     * Correlation ID can be provided when creating a command. All events emitted during the command processing
     * will contain the same correlation ID.
     *
     * If the correlation ID is not specified, a random string will be assigned.
     */
    correlationId?: string;
}

//
//
//

/**
 * Loads dashboard from analytical backend.
 *
 * @internal
 */
export interface LoadDashboard extends IDashboardCommand {
    type: "GDC.DASHBOARD.CMD.LOAD";
    payload: {
        config?: DashboardConfig;
        permissions?: IWorkspacePermissions;
    };
}

/**
 * Creates the LoadDashboard command.
 *
 * @param config - dashboard configuration provided by the user. if not specified, the load command WILL load
 *  the configuration from backend.
 * @param correlationId - optionally specify correlation id to use for this command
 * @internal
 */
export function loadDashboard(config?: DashboardConfig, correlationId?: string): LoadDashboard {
    return {
        type: "GDC.DASHBOARD.CMD.LOAD",
        correlationId,
        payload: {
            config,
        },
    };
}

//
//
//

/**
 * @internal
 */
export type DashboardCommands = LoadDashboard;
