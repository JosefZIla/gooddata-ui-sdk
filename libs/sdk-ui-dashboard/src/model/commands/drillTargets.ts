// (C) 2021 GoodData Corporation

import { ObjRef } from "@gooddata/sdk-model";
import { IAvailableDrillTargets } from "@gooddata/sdk-ui";
import { IDashboardCommand } from "./base";

/**
 * Add widget drill targets
 *
 * @alpha
 */
export interface AddDrillTargets extends IDashboardCommand {
    readonly type: "GDC.DASH/CMD.DRILL_TARGETS.ADD";
    readonly payload: {
        readonly ref: ObjRef;
        readonly availableDrillTargets: IAvailableDrillTargets;
    };
}

/**
 * Create AddDrillTargets {@link AddDrillTargets} command.
 *
 * @param ref - Unique widget ref
 * @param availableDrillTargets - Available widget drill targets {@link @gooddata/sdk-ui#IAvailableDrillTargets}
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 * @returns AddDrillTargets command
 *
 * @alpha
 */
export function addDrillTargets(
    ref: ObjRef,
    availableDrillTargets: IAvailableDrillTargets,
    correlationId?: string,
): AddDrillTargets {
    return {
        type: "GDC.DASH/CMD.DRILL_TARGETS.ADD",
        correlationId,
        payload: {
            ref,
            availableDrillTargets,
        },
    };
}
