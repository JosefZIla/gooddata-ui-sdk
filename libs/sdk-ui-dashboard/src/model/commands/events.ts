// (C) 2021-2022 GoodData Corporation

import { DashboardEventBody, ICustomDashboardEvent, IDashboardEvent } from "../events/base";
import { IDashboardCommand } from "./base";

/**
 * Payload of the {@link TriggerEvent} command.
 * @alpha
 */
export interface TriggerEventPayload {
    /**
     * Event body without the {@link DashboardContext} property. That will be filled when the command is processed.
     */
    readonly eventBody: DashboardEventBody<IDashboardEvent | ICustomDashboardEvent>;
}

/**
 * Triggers an event.
 *
 * @alpha
 */
export interface TriggerEvent extends IDashboardCommand {
    readonly type: "GDC.DASH/CMD.EVENT.TRIGGER";
    readonly payload: TriggerEventPayload;
}

/**
 * Creates an {@link TriggerEvent} command.
 *
 * @alpha
 */
export function triggerEvent(
    eventBody: DashboardEventBody<IDashboardEvent | ICustomDashboardEvent>,
    correlationId?: string,
): TriggerEvent {
    return {
        type: "GDC.DASH/CMD.EVENT.TRIGGER",
        correlationId,
        payload: {
            eventBody,
        },
    };
}
