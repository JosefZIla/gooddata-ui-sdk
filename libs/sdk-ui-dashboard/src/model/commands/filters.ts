// (C) 2021 GoodData Corporation
import { IDashboardAttributeFilterParent } from "@gooddata/sdk-backend-spi";
import { IAttributeElements, ObjRef } from "@gooddata/sdk-model";
import { IDashboardCommand } from "./base";

/**
 * @internal
 */
export interface AddAttributeFilter extends IDashboardCommand {
    readonly type: "GDC.DASHBOARD.CMD.AF.ADD";
    readonly payload: {
        readonly displayForm: ObjRef;

        readonly index: number;

        /**
         * Optionally specify parent filters whose selected values will be used to narrow
         * down the selection in this newly added filter.
         *
         * XXX: not needed in the initial version; would be good for API completeness
         */
        readonly parentFilters?: ReadonlyArray<IDashboardAttributeFilterParent>;

        /**
         * Optionally specify the initial selection of attribute elements. If not provided all
         * elements will be selected by default.
         *
         * XXX: not needed in the initial version; would be good for API completeness
         */
        readonly initialSelection?: IAttributeElements;
    };
}

/**
 * Creates the AddAttributeFilter command. Dispatching this command will result in the addition
 * of another attribute filter to the dashboard's filter bar, at desired position.
 *
 * The filter will be set for the display form provided by reference. When created, the filter will be
 * no-op - all the elements will be selected.
 *
 * @param displayForm - specify attribute display form which will be used for filtering
 * @param index - specify index among the attribute filters at which the new filter should be placed.
 *  The index starts at zero and there is convenience that index of -1 would add the filter at the end.
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 * @internal
 */
export function addAttributeFilter(
    displayForm: ObjRef,
    index: number,
    correlationId?: string,
): AddAttributeFilter {
    return {
        type: "GDC.DASHBOARD.CMD.AF.ADD",
        correlationId,
        payload: {
            displayForm,
            index,
        },
    };
}

//
//
//

/**
 * @internal
 */
export interface RemoveAttributeFilters extends IDashboardCommand {
    readonly type: "GDC.DASHBOARD.CMD.AF.REMOVE";
    readonly payload: {
        /**
         * XXX: we do not necessarily need to remove multiple filters atm, but this should
         *  be very easy to do and adds some extra flexibility.
         */
        readonly filterLocalId: string[];
    };
}

/**
 * Creates the RemoveAttributeFilters command. Dispatching this command will result in the removal
 * of dashboard's attribute filter with the provided local identifier.
 *
 * @param filterLocalId - dashboard attribute filter's local identifier
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 * @internal
 */
export function removeAttributeFilter(filterLocalId: string, correlationId?: string): RemoveAttributeFilters {
    return {
        type: "GDC.DASHBOARD.CMD.AF.REMOVE",
        correlationId,
        payload: {
            filterLocalId: [filterLocalId],
        },
    };
}

//
//
//

/**
 * @internal
 */
export interface MoveAttributeFilter extends IDashboardCommand {
    readonly type: "GDC.DASHBOARD.CMD.AF.MOVE";
    readonly payload: {
        readonly filterLocalId: string;
        readonly index: number;
    };
}

/**
 * Creates the MoveAttributeFilter command. Dispatching this command will result in move of the dashboard attribute
 * filter with the provided local id to a new spot. The new spot is defined by index. For convenience the index
 * of -1 means move to the end of the attribute filter list.
 *
 * @param filterLocalId - dashboard filter's local identifier
 * @param index - specify index among the attribute filters at which the new filter should be placed.
 *  The index starts at zero and there is convenience that index of -1 would add the filter at the end.
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 *
 * @internal
 */
export function moveAttributeFilter(
    filterLocalId: string,
    index: number,
    correlationId?: string,
): MoveAttributeFilter {
    return {
        type: "GDC.DASHBOARD.CMD.AF.MOVE",
        correlationId,
        payload: {
            filterLocalId,
            index,
        },
    };
}

//
//
//

/**
 * @internal
 */
export type AttributeFilterSelectionType = "IN" | "NOT_IN";

/**
 * @internal
 */
export interface ChangeAttributeFilterSelection extends IDashboardCommand {
    readonly type: "GDC.DASHBOARD.CMD.AF.CHANGE_SELECTION";
    readonly payload: {
        readonly filterLocalId: string;
        readonly elements: IAttributeElements;
        readonly selectionType: AttributeFilterSelectionType;
    };
}

/**
 * Creates the ChangeAttributeFilterSelection command. Dispatching this command will result in application
 * of element selection for the dashboard attribute filter with the provided id.
 *
 * The attribute elements can be provided either using their URI (primary key) or value. Together with the
 * elements you must indicate the selection type - either 'IN' or 'NOT_IN'.
 *
 * @remarks see {@link resetAttributeFilterSelection} for convenience function to select all attribute elements of
 *  particular filter.
 *
 * @param filterLocalId - dashboard attribute filter's local id
 * @param elements - elements
 * @param selectionType - selection type. either 'IN' or 'NOT_IN'
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 *
 * @internal
 */
export function changeAttributeFilterSelection(
    filterLocalId: string,
    elements: IAttributeElements,
    selectionType: AttributeFilterSelectionType,
    correlationId?: string,
): ChangeAttributeFilterSelection {
    return {
        type: "GDC.DASHBOARD.CMD.AF.CHANGE_SELECTION",
        correlationId,
        payload: {
            filterLocalId,
            elements,
            selectionType,
        },
    };
}

/**
 * A convenience function that will create ChangeAttributeFilterSelection command that will select all
 * elements of the dashboard attribute filter with the provided local id.
 *
 * This is same as creating the ChangeAttributeFilterSelection command with empty elements and NOT_IN selection type.
 *
 * @param filterLocalId - dashboard attribute filter's local id
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 *
 * @internal
 */
export function resetAttributeFilterSelection(
    filterLocalId: string,
    correlationId?: string,
): ChangeAttributeFilterSelection {
    return {
        type: "GDC.DASHBOARD.CMD.AF.CHANGE_SELECTION",
        correlationId,
        payload: {
            filterLocalId,
            elements: { uris: [] },
            selectionType: "NOT_IN",
        },
    };
}

//
//
//

/**
 * @internal
 */
export interface SetAttributeFilterParent extends IDashboardCommand {
    readonly type: "GDC.DASHBOARD.CMD.AF.SET_PARENT";
    readonly payload: {
        readonly filterLocalId: string;
        readonly parentFilters: ReadonlyArray<IDashboardAttributeFilterParent>;
    };
}

/**
 * Creates the SetAttributeFilterParent command. Dispatching this command will result in setting a parent-child
 * relationship between two filters.
 *
 * When an attribute filter has a parent set up, the attribute elements that will be available in the child
 * filter will be influenced by the selection in the parent. Only elements which have a link over some attributes
 * to the parent filter will be available for selection in the child filter.
 *
 * TODO: fix the docs
 *
 * @param filterLocalId - local id of filter that will be a child in the relationship
 * @param parentFilter - definition of the relationship to parent, this contains local id of the parent filter and
 *  one or more 'over' attributes. The 'over' attributes will be included when querying
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 *
 * @internal
 */
export function setAttributeFilterParent(
    filterLocalId: string,
    parentFilter: IDashboardAttributeFilterParent,
    correlationId?: string,
): SetAttributeFilterParent {
    return {
        type: "GDC.DASHBOARD.CMD.AF.SET_PARENT",
        correlationId,
        payload: {
            filterLocalId,
            parentFilters: [parentFilter],
        },
    };
}
