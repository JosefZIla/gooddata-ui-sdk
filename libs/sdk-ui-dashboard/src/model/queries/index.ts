// (C) 2021 GoodData Corporation

import { QueryInsightAttributesMeta, QueryInsightDateDatasets } from "./insights";
import { QueryWidgetFilters } from "./widgets";

export { IDashboardQuery, DashboardQueryType } from "./base";
export {
    QueryInsightDateDatasets,
    InsightDateDatasets,
    queryDateDatasetsForInsight,
    QueryInsightAttributesMeta,
    InsightAttributesMeta,
    queryInsightAttributesMeta,
    insightSelectDateDataset,
} from "./insights";
export { QueryMeasureDateDatasets, queryDateDatasetsForMeasure, MeasureDateDatasets } from "./kpis";
export { QueryWidgetFilters as QueryInsightWidgetFilters, queryWidgetFilters } from "./widgets";

/**
 * @alpha
 */
export type DashboardQueries = QueryInsightDateDatasets | QueryInsightAttributesMeta | QueryWidgetFilters;
