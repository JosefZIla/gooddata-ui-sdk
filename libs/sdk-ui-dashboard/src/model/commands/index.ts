// (C) 2021 GoodData Corporation

import { LoadDashboard, RenameDashboard, ResetDashboard, SaveDashboard, SaveDashboardAs } from "./dashboard";
import {
    AddAttributeFilter,
    ChangeAttributeFilterSelection,
    ChangeDateFilterSelection,
    MoveAttributeFilter,
    RemoveAttributeFilters,
    SetAttributeFilterParent,
} from "./filters";

export { DashboardCommandType, IDashboardCommand } from "./base";
export {
    LoadDashboard,
    loadDashboard,
    SaveDashboardAs,
    saveDashboardAs,
    SaveDashboard,
    saveDashboard,
    RenameDashboard,
    renameDashboard,
    ResetDashboard,
    resetDashboard,
} from "./dashboard";
export {
    ChangeDateFilterSelection,
    changeDateFilterSelection,
    clearDateFilterSelection,
    AddAttributeFilter,
    addAttributeFilter,
    MoveAttributeFilter,
    moveAttributeFilter,
    RemoveAttributeFilters,
    removeAttributeFilter,
    ChangeAttributeFilterSelection,
    AttributeFilterSelectionType,
    resetAttributeFilterSelection,
    changeAttributeFilterSelection,
    SetAttributeFilterParent,
    setAttributeFilterParent,
} from "./filters";

/**
 * @internal
 */
export type DashboardCommands =
    | LoadDashboard
    | SaveDashboard
    | SaveDashboardAs
    | RenameDashboard
    | ResetDashboard
    | ChangeDateFilterSelection
    | AddAttributeFilter
    | RemoveAttributeFilters
    | MoveAttributeFilter
    | ChangeAttributeFilterSelection
    | SetAttributeFilterParent;
