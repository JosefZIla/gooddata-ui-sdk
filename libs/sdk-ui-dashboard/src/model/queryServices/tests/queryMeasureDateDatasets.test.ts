// (C) 2021 GoodData Corporation
import { DashboardTester } from "../../tests/DashboardTester";
import {
    EmptyDashboardIdentifier,
    MockAvailabilityWithDifferentRelevance,
    MockAvailabilityWithSameRelevance,
} from "../../tests/Dashboard.fixtures";
import { MeasureDateDatasets, queryDateDatasetsForMeasure } from "../../queries";
import { measureItem } from "@gooddata/sdk-model";
import { ICatalogDateDataset } from "@gooddata/sdk-backend-spi";
import { loadDashboard } from "../../commands";
import { ReferenceLdm } from "@gooddata/reference-workspace";

function datasetsDigest(
    datasets: ReadonlyArray<ICatalogDateDataset | undefined>,
): ReadonlyArray<string | undefined> {
    return datasets.map((d) => d?.dataSet.title);
}

describe("query measure date datasets", () => {
    describe("for datasets obtained from catalog", () => {
        // given all available datasets, this availability mock will pick 2 by name and associate relevance so that second one has more relevance

        it("should order date datasets by relevance desc", async () => {
            const Tester = DashboardTester.forRecording(
                EmptyDashboardIdentifier,
                {},
                {
                    catalogAvailability: {
                        availableDateDatasets: MockAvailabilityWithDifferentRelevance,
                    },
                },
            );

            await Tester.dispatchAndWaitFor(loadDashboard(), "GDC.DASH/EVT.LOADED");

            const result: MeasureDateDatasets = await Tester.query(
                queryDateDatasetsForMeasure(measureItem(ReferenceLdm.Won)),
            );

            expect(datasetsDigest(result.dateDatasetsOrdered)).toMatchSnapshot();
            expect(result.dateDatasetDisplayNames).toMatchSnapshot();
        });

        it("should order date datasets by relevance and title if tied", async () => {
            const Tester = DashboardTester.forRecording(
                EmptyDashboardIdentifier,
                {},
                {
                    catalogAvailability: {
                        availableDateDatasets: MockAvailabilityWithSameRelevance,
                    },
                },
            );

            await Tester.dispatchAndWaitFor(loadDashboard(), "GDC.DASH/EVT.LOADED");

            const result: MeasureDateDatasets = await Tester.query(
                queryDateDatasetsForMeasure(measureItem(ReferenceLdm.Won)),
            );

            expect(datasetsDigest(result.dateDatasetsOrdered)).toMatchSnapshot();
            expect(result.dateDatasetDisplayNames).toMatchSnapshot();
        });
    });
});
