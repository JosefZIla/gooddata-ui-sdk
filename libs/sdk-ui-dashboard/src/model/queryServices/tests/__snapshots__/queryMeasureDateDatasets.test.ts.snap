// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query measure date datasets for datasets obtained from catalog should order date datasets by relevance and title if tied 1`] = `
Array [
  "Date (Activity)",
  "Date (Timeline)",
]
`;

exports[`query measure date datasets for datasets obtained from catalog should order date datasets by relevance and title if tied 2`] = `
Object {
  "Date (Activity)": "Activity",
  "Date (Timeline)": "Timeline",
}
`;

exports[`query measure date datasets for datasets obtained from catalog should order date datasets by relevance desc 1`] = `
Array [
  "Date (Timeline)",
  "Date (Activity)",
]
`;

exports[`query measure date datasets for datasets obtained from catalog should order date datasets by relevance desc 2`] = `
Object {
  "Date (Activity)": "Activity",
  "Date (Timeline)": "Timeline",
}
`;
